autoload/ale/fixers/prettier.vim

```diff
@@ -39,6 +39,21 @@ function! ale#fixers#prettier#Fix(buffer) abort
     let l:use_config = ale#Var(a:buffer, 'javascript_prettier_use_local_config')
                 \ && !empty(l:config)
 
+    " Append the --parser flag depending on the current filetype (unless it's
+    " already set in g:javascript_prettier_options).
+    if match(l:options, '--parser') == -1
+        if &filetype is# 'typescript'
```

Comment:
You should use `getbufvar` for `&filetype`. Because of the way async execution works, this code could be run from another buffer.

Reply:
Ah, right! This ought to do it 829e0442d191c711ed537c2c18a56b67d86a72e2